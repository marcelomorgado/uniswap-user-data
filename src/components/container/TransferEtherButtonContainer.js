import React, { Fragment } from "react";
import TransferEtherModal from "../presentational/TransferEtherModal";
import TransferEtherButton from "../presentational/TransferEtherButton";
import { ApolloConsumer } from "react-apollo";
import { SEND_ETHER } from "../../queries";

class TransferEtherButtonContainer extends React.Component {
  state = {
    modalOpen: false,
  };

  openModal = () => {
    this.setState({ modalOpen: true });
  };

  closeModal = () => {
    this.setState({ modalOpen: false });
  };

  render() {
    const { state, openModal, closeModal } = this;
    const { modalOpen } = state;
    const onClick = () => {
      this.openModal();
    };

    return (
      <Fragment>
        <TransferEtherButton onClick={onClick} />

        <ApolloConsumer>
          {client => {
            const handleTransfer = async (from, to) => {
              // client.mutate({
              //   mutation: UPDATE_USER,
              //   variables: {
              //     userId: "0x0000000000000000000000000000000000000000",
              //     etherBalance: "123",
              //   },
              // });

              console.log(`${from} -> ${to}`);

              client.mutate({
                mutation: SEND_ETHER,
                variables: {
                  from,
                  to,
                  amount: "1",
                },
              });
            };
            return (
              <TransferEtherModal
                open={modalOpen}
                handleOpen={openModal}
                handleClose={closeModal}
                handleTransfer={handleTransfer}
              />
            );
          }}
        </ApolloConsumer>
      </Fragment>
    );
  }
}

export default TransferEtherButtonContainer;
