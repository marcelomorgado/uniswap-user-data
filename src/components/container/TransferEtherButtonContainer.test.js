import React from "react";
import { shallow } from "enzyme";
import TransferEtherButtonContainer from "./TransferEtherButtonContainer";

it("renders the component", () => {
  expect(shallow(<TransferEtherButtonContainer />)).toMatchSnapshot();
});
