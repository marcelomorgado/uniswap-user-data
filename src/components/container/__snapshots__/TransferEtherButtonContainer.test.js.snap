// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Fragment>
  <WithStyles(TransferEtherButton)
    onClick={[Function]}
  />
  <ApolloConsumer>
    <Component />
  </ApolloConsumer>
</Fragment>
`;
